ansible/roles/vagrant_local/tasks/main.yml

```diff
@@ -1,10 +1,6 @@
 ---
-- name: Set the hostname in /etc/hostname
-  shell: echo {{ vagrant_local.vm.hostname }} > /etc/hostname
```

Comment:
Use the appropriate ansible module ([`hostname`](http://docs.ansible.com/ansible/hostname_module.html)) to set the hostname should make this step not required.